packages/app/src/specs/SpecsListBanners.vue

```diff
@@ -175,4 +237,41 @@ watch(
   },
 )
 
+const cloudData = computed(() => ([props.gql.cloudViewer, props.gql.currentProject] as const))
+
+/**
+ * It takes a non-zero amount of time for cloud data to be fetched, during which we can't reliably tell whether the user is really
+ * logged in/has an org/etc. Here we attempt to delay this watcher to give the cloud GQL a little time to resolve before deciding
+ * which banners to display. Not doing this causes us to immediately show the "login" banner then have it disappear a second later
+ * once login state resolves
+ */
+debouncedWatch(
+  cloudData,
+  ([cloudViewer, currentProject]) => {
+    const isLoggedIn = !!cloudViewer?.id
+    // Need to be able to tell whether the lack of `firstOrganization` means they don't have an org or whether it just hasn't loaded yet
+    // Not having this check can cause a brief flicker of the 'Create Org' banner while org data is loading
+    const isOrganizationLoaded = !!cloudViewer?.firstOrganization
+    const isMemberOfOrganization = (cloudViewer?.firstOrganization?.nodes?.length ?? 0) > 0
```

Comment:
Just to keep the thoughts around making GraphQL more readable front-and-center of mind ... I think we might want to start exploring some kind of pattern to model this logic, hopefully independently of the component that consumes it. 

I wonder if we can rip this logic out and put it into a separate model that we could potentially test in isolation? It's pretty hard to grok the component when there's all this complexity sitting inside a giant `watch`. Some ideas that come to mind for what might be easier to understand in the future:

```js
const { showRecordBanner } = useBanners(props.gql)
// or
const bannerModel = createBannerModel(props.gql.currentProject)

watch(bannerModel, newVal => {
  showRecordBanner.value = newVal.showRecordBanner
})
```

When I did React/Redux, this separation was more simple with `connect` used to separate container/components. With the move towards hooks/composables, it's not so clear anymore.

Just sparking some discussion - what does you think? If we are happy with the "put it all in the component" pattern, that's fine, too. Commenting primarily to ensure we keep thinking about this as we learn more about how best to utilise GraphQL as we more deeply integrate the Cloud and App, since I do see this getting out of hand eventually.

Reply:
Absolutely agree, I don't feel this is elegant or a long-term solution by any means. The reason I went this way is that there is an upcoming epic (#23072) to build a framework for user messaging in which a more holistic pattern for all our banners will be constructed. It will handle hooking up to state, managing preconditions, A/B testing, etc

Reply:
Sounds good, let's deal with it there.